src/Datadog.Trace/ExtensionMethods/StringExtensions.cs

```diff
@@ -22,31 +22,47 @@ public static string TrimEnd(this string value, string suffix, StringComparison
 
         /// <summary>
         /// Converts a <see cref="string"/> into a <see cref="bool"/> by comparing it to commonly used values
-        /// such as "True", "yes", or "1". Case-insensitive. Defaults to <c>false</c> if string is not recognized.
+        /// such as "True", "yes", "T", "Y", or "1" for <c>true</c> and "False", "no", "F", "N", or "0" for <c>false</c>. Case-insensitive.
+        /// Defaults to <c>null</c> if string is not recognized.
         /// </summary>
         /// <param name="value">The string to convert.</param>
-        /// <returns><c>true</c> if <paramref name="value"/> is one of the accepted values for <c>true</c>; <c>false</c> otherwise.</returns>
+        /// <returns><c>true</c> or <c>false</c> if <paramref name="value"/> is one of the accepted values; <c>null</c> otherwise.</returns>
```

Comment:
Thanks for updating the docs! This used to be true at some point, but then we changed the method to return `bool?` instead of `bool`.

Reply:
Yeah, It was the intent for the PR, based in a comment from @zacharycmontoya in other PR, but then I thought I can improve the implementation by removing the allocation.